pkg/tekton/metapipeline/metapipeline.go

```diff
@@ -216,14 +215,15 @@ func stepCreateTektonCRDs(params CRDCreationParameters) syntax.Step {
 	args = append(args, "--service-account", params.ServiceAccount)
 	args = append(args, "--source", params.SourceDir)
 	args = append(args, "--branch", params.BranchIdentifier)
+	args = append(args, "--build-number", params.BuildNumber)
 	if params.Context != "" {
 		args = append(args, "--context", params.Context)
 	}
-	if len(params.Labels) > 0 {
-		args = append(args, "--label", strings.Join(params.Labels, ","))
+	for _, l := range params.Labels {
+		args = append(args, "--label", l)
 	}
-	if len(params.EnvVars) > 0 {
-		args = append(args, "--env", strings.Join(params.EnvVars, ","))
+	for _, e := range params.EnvVars {
```

Comment:
This is ways better. Good catch, using the flags multiple times is the better approach.